back-end/src/sciler/config/checkConfig.go

```diff
@@ -0,0 +1,367 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// checkConfig is a method that will return an error
+// if the constraints value type is not equal to the device input type specified,
+// the actions type is not equal to the device output type,
+// or some other not allowed json configuration
+func checkConfig(config WorkingConfig) []string {
+	errList := make([]string, 0)
+	for _, puzzle := range config.Puzzles {
+		for _, rule := range puzzle.Event.Rules {
+			if err := rule.Conditions.checkConstraints(config, rule.ID); err != nil {
+				errList = append(errList, err...)
+			}
+			errList = append(errList, checkActions(rule.Actions, config)...)
+		}
+	}
+
+	for _, generalEvent := range config.GeneralEvents {
+		for _, rule := range generalEvent.Rules {
+			if err := rule.Conditions.checkConstraints(config, rule.ID); err != nil {
+				errList = append(errList, err...)
+			}
+			errList = append(errList, checkActions(rule.Actions, config)...)
+		}
+	}
+
+	for _, rule := range config.ButtonEvents {
+		if err := rule.Conditions.checkConstraints(config, rule.ID); err != nil {
+			errList = append(errList, err...)
+		}
+		errList = append(errList, checkActions(rule.Actions, config)...)
+	}
+
+	// TODO check uniqueness of all device_id, timer_id and rule_id
```

Comment:
not a comment on your MR but something we need to do, we probably do not want to hand in code base with TODO's

Reply:
I made an issue for this todo: #191